File: adventofcode/day4/day4.py

with open('day4Input.txt') as f:
    Lines = f.readlines()

count = 0

for line in Lines:
    if line == "\n": break
    a, b = line.split(",")
    x, y = a.split("-")
    x1, y1 = b.split("-")

    if   int(x1) <= int(x) <= int(y1): count += 1
    elif int(x1) <= int(y) <= int(y1): count += 1
    elif int(x)  <= int(x1)<= int(y): count += 1
    elif int(x)  <= int(y1)<= int(y): count += 1

print(count)

File: adventofcode/day4/day4.hs
main = do
  content <- readFile "day4Input.txt"
  let tupleRanges = parseInput (lines content)
  putStrLn $ show $ countOverlaps tupleRanges

countOverlaps :: [(Int,Int,Int,Int)] -> Int
countOverlaps [] = 0
countOverlaps ((x,y,x1,y1):rest) = (overlap x y x1 y1) + countOverlaps rest

overlap :: Int -> Int -> Int -> Int -> Int
overlap x y x1 y1
  | x >= x1 && y <= y1 = 1
  | x <= x1 && y >= y1 = 1
  | otherwise = 0


parseInput :: [String] -> [(Int,Int,Int,Int)]
parseInput ("":_) = []
parseInput ((x:'-':y:',':x1:'-':y1):rest) = (read x, read y,read x1,read y1) : parseInput rest
